pavelib/prereqs.py

```diff
@@ -164,14 +164,83 @@ def install_node_prereqs():
 
 
 @task
+def uninstall_obsolete_python_packages():
+    """
+    Uninstall Python packages that need explicit uninstallation.
+
+    Some Python packages that we no longer want need to be explicitly
+    uninstalled, notably, South.  Some other packages were once installed in
+    ways that were resistant to being upgraded, like edxval.  Also uninstall
+    them.
+
+    """
+    # So that we don't constantly uninstall things, use a version number of the
+    # uninstallation needs.  Check it, and skip this if we're up to date.
+    VERSION = 1
+    state_file_path = os.path.join(PREREQS_STATE_DIR, "python_uninstall_version.txt")
+    if os.path.isfile(state_file_path):
+        with open(state_file_path) as state_file:
+            version = int(state_file.read())
+        if version == VERSION:
+            return
+
+    # Run pip to find the packages we need to get rid of.  Believe it or not,
+    # edx-val is installed in a way that it is present twice, so we have a loop
+    # to really really get rid of it.
+    for attempt in range(3):
+        uninstalled = False
+        frozen = sh("pip freeze", capture=True).splitlines()
+
+        # Uninstall South
+        if any(line.startswith("South") for line in frozen):
+            sh("pip uninstall -y South")
+            uninstalled = True
+
+        # Uninstall edx-val
+        if any(line.startswith("-e ") and "edxval" in line for line in frozen):
```

Comment:
Hmm, good point.  The -e check was originally a way to avoid always uninstalling and re-installing it, and the fixed version of edxval is not -e-installed, but now that we check the version of uninstall we're on, that isn't an issue.
I'll update it.